src/server.rs

```diff
@@ -283,10 +285,78 @@ async fn upload(
     Ok(HttpResponse::Ok().body(urls.join("")))
 }
 
+/// File entry item for list endpoint.
+#[derive(Serialize, Deserialize)]
+pub struct ListItem {
+    /// Uploaded file name.
+    pub filename: String,
+    /// Size of the file in bytes.
+    pub filesize: u64,
+    /// ISO8601 formatted date-time string of the expiration timestamp if one exists for this file.
+    pub expires_at: Option<String>,
+}
+
+/// Returns the list of files.
+#[get("/list")]
+async fn list(
+    request: HttpRequest,
+    config: web::Data<RwLock<Config>>,
+) -> Result<HttpResponse, Error> {
+    let config = config
+        .read()
+        .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?
+        .clone();
+    let connection = request.connection_info().clone();
+    let host = connection.realip_remote_addr().unwrap_or("unknown host");
+
+    let tokens = config.get_tokens();
+    auth::check(host, request.headers(), tokens)?;
```

Comment:
In order to _disable_ this feature when no authorization is set, you need to check the existence of `tokens` (and return an error such as "authorization is not set") instead of the normal authentication flow.

Right now the files are served if `expose_list` is set to `true`, regardless of authentication tokens.

Reply:
Isn't this is the same flow as for `/version`?

I also tested this manually and it works as expected.

Reply:
I think this is pretty much the last item to clarify. Or is there anything else?

Maybe I misunderstood your concern, but I can't seem to find an issue with the current implementation. I also listed the test cases above. I haven't uploaded any files, thus the output for a successful request/respones is just `[]`. (At the beginning of the line, since there's no newline at the end of the json output.)

Reply:
I meant disabling the `list` endpoint completely when no tokens is set but with a second thought, it would be better if we follow the same logic as `version` endpoint.